lib/jormungandr/test/integration/Cardano/Pool/MetricsSpec.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeApplications #-}
+module Cardano.Pool.MetricsSpec where
+
+import Prelude
+
+import Cardano.BM.Configuration.Static
+    ( defaultConfigStdout )
+import Cardano.BM.Trace
+    ( nullTracer )
+import Cardano.Pool.Metrics
+    ( monitorStakePools )
+import Cardano.Wallet.Jormungandr
+    ( toSPBlock )
+import Cardano.Wallet.Jormungandr.Network
+    ( JormungandrBackend (..)
+    , JormungandrConfig (..)
+    , withJormungandr
+    , withNetworkLayer
+    )
+import Cardano.Wallet.Primitive.Types
+    ( BlockHeader (..) )
+import Control.Concurrent
+    ( forkIO, killThread, threadDelay )
+import Control.Exception
+    ( throwIO )
+import Control.Monad
+    ( void )
+import Data.Text
+    ( Text )
+import Fmt
+    ( pretty )
+import System.IO.Temp
+    ( withSystemTempDirectory )
+import System.Process
+    ( StdStream (..) )
+import Test.Hspec
+    ( Spec, around, it, shouldBe )
+import Test.Integration.Framework.DSL
+    ( eventually )
+
+import qualified Cardano.Pool.DB as Pool
+import qualified Cardano.Pool.DB.Sqlite as Pool
+
+spec :: Spec
+spec = around startNode $ do
+    it "a monitorStakePools thread with db, if killed and restarted, catches up\
+        \ with one that stays running" $ \(nl, referenceDB) ->
+        withDB "/db.sqlite" $ \db -> do
+            referenceWorker <- fork nl referenceDB
+            worker <- fork nl db
+
+            threadDelay (5 * second)
+            killThread worker
+            threadDelay (5 * second)
+            worker' <- fork nl db
+
+            eventually $ do
+                r1 <- showStoredSlots referenceDB
+                r2 <- showStoredSlots db
+                r1 `shouldBe` r2
+
+            killThread worker'
+            killThread referenceWorker
```

Comment:
If the assertion fails, this will leak worker threads. You should consider using `before` & `after` to correctly setup and teardown the relevant threads around the spec itself